dimelo/utils.py

```diff
@@ -7,16 +7,59 @@
 from matplotlib.axes import Axes
 
 # This provides the mapping of canonical bases to sets of valid mode names
-BASEMOD_NAMES_DICT = {
-    "A": {"A", "a", "Y"},
-    "C": {"m", "Z"},
-}
-
-DEFAULT_COLORS = {
-    "A,0": "blue",
-    "CG,0": "orange",
-    "GCH,1": "purple",
-}
+BASEMOD_NAMES_DICT = defaultdict(lambda: set())
+BASEMOD_NAMES_DICT.update(
+    {
+        "A": {"a", "Y"},
+        "C": {"m", "Z"},
+    }
+)
+
+DEFAULT_COLORS = defaultdict(lambda: "grey")
+DEFAULT_COLORS.update(
+    {
+        "A,0": "blue",
+        "A,0,a": "blue",
+        "CG,0": "orange",
+        "CG,0,m": "yellow",
+        "CG,0,h": "red",
+        "GCH,1": "purple",
+    }
+)
+
+
+class ParsedMotif:
+    def __init__(self, motif_string):
```

Comment:
Can you add a brief docstring to `__init__` describing the attributes of this class and what it represents?

Reply:
Also, this could be a `dataclass` or `namedtuple`, since it doesn't have any methods, but we haven't talked about that yet so let's table that discussion until Thursday at the earliest.